models/Element/Service.php

```diff
@@ -1112,11 +1112,16 @@ public static function getValidKey($key, $type)
         $key = $event->getArgument('key');
         $key = trim($key);
 
+        // replace all control/unassigned and invalid characters
+        $key = preg_replace('/[^\PCc^\PCn^\PCs]/u', '', $key);
         // replace all 4 byte unicode characters
         $key = preg_replace('/[\x{10000}-\x{10FFFF}]/u', '-', $key);
         // replace slashes with a hyphen
         $key = str_replace('/', '-', $key);
 
+        // replace some other special characters
+        $key = preg_replace('/[\t\n\r]/', '', $key);
```

Comment:
What about the other whitespace characters `\f` (form-feed) and `\v` (vertical whitespace)

Reply:
👍 added - thanks for the hint!